### PR TITLE
Bcrypt implementation

### PR DESCRIPTION
**Pull request by 1Conan**
Needs testing. 
Sorry for the additional commit to change installer.sh back. I can't find a revert button on the iOS git client I use. 

As for moving the session_start(), I'm sure that it's good practice to put that on the very top of the code. (I think you should move the others too)

### CODE REVIEW

**Comment by 1Conan** [comment]
Automatic bcrypt rounds calculation is implemented :D 
So it would be more secure as long as a more powerful system is used. 
Salt is automatically generated and we don't need an additional column for salts.  (Also itms a per user salt yay more secure :D)

**Comment by FlamesRunner** [comment]
Awesome :)

Now, I need to go test it, so please be patient on that.

File: installer.sh [hunk 1]
```diff
@@ -160,7 +161,7 @@ sleep 3
 
 clear
 
-yourpubip=`curl -q -s icanhazip.com`
+yourpubipv4=`curl -q -s ipv4.icanhazip.com`
```
Comment (1Conan) [inline installer.sh]: I might add a check for ipv6.  
Better be specific :D

**Comment by 1Conan** [comment]
Any updates?

**Comment by FlamesRunner** [comment]
Yep. I have merged the pull request.

**Pull request merged**